includes/amp-helper-functions.php

```diff
@@ -512,43 +571,40 @@ function amp_add_amphtml_link() {
 
 	$current_url = amp_get_current_url();
 
-	$amp_url = null;
-	if ( current_theme_supports( AMP_Theme_Support::SLUG ) ) {
-		if ( AMP_Theme_Support::is_paired_available() ) {
-			$amp_url = add_query_arg( amp_get_slug(), '', $current_url );
-		}
-	} elseif ( $wp_query instanceof WP_Query && ( $wp_query->is_singular() || $wp_query->is_posts_page ) && post_supports_amp( get_post( get_queried_object_id() ) ) ) {
-		$amp_url = amp_get_permalink( get_queried_object_id() );
-	}
-
-	if ( ! $amp_url ) {
-		printf( '<!-- %s -->', esc_html__( 'There is no amphtml version available for this URL.', 'amp' ) );
-		return;
-	}
-
 	// Check to see if there are known unaccepted validation errors for this URL.
 	if ( current_theme_supports( AMP_Theme_Support::SLUG ) ) {
```

Comment:
Eventually this logic here should probably be moved inside of `is_amp_available()`, but we should check the implications of doing so first.

Reply:
To examine later.